src/query/v2/multiframe.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "query/v2/multiframe.hpp"
+
+#include "query/v2/bindings/frame.hpp"
+#include "utils/pmr/vector.hpp"
+
+namespace memgraph::query::v2 {
+
+MultiFrame::MultiFrame(FrameWithValidity default_frame, size_t number_of_frames,
+                       utils::MemoryResource *execution_memory)
+    : default_frame_(default_frame),
+      frames_(utils::pmr::vector<FrameWithValidity>(number_of_frames, default_frame, execution_memory)) {
+  MG_ASSERT(number_of_frames > 0);
+  MG_ASSERT(!default_frame.IsValid());
+}
+
+MultiFrame::MultiFrame(const MultiFrame &other) : default_frame_(other.default_frame_) {
+  /*
+  TODO
+  Do we just copy all frames or do we make distinctions between valid and not valid frames? Does it make any
+  difference?
```

Comment:
I think we should just copy and don't care about the valid/invalid state.

Reply:
I have also removed default_frame, if we don't care about valid/invalid state there's no need to keep it